wireapp/wire-android
# fix: showing download confirmation snackbar message after saving assets [AR-1729]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1729" title="AR-1729" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />AR-1729</a>  [download/save assets] "success" toast is sometimes shown behind the dialog
  </td></table>
  <br />
 


----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues
After downloading some asset correctly, the toast message was being shown only once because we were clearing the state view on the wrong VM. With this fix, after displaying a message, the snackbar messages view states values will be cleared on both `MessageComposerViewModel` and `ConversationMessagesViewModel`

#### How to Test

- Receive/Send an asset. 
- Click on Save asset.
- See how the snackbar message is displayed.
- Repeat the process and check the toast appears again.

### Attachments (Optional)
https://user-images.githubusercontent.com/2468164/209695082-6b8905db-7e32-4a5b-a40d-7573946f640c.mp4 https://user-images.githubusercontent.com/2468164/209695110-11b826c2-53f3-41ec-9105-f2ae94df0c81.mp4

----

#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
